### PR TITLE
Ports: Add RVVM port

### PR DESCRIPTION
**Pull request by LekKit**
This PR brings [RVVM](https://github.com/LekKit/RVVM) port, allowing *fast* RISC-V emulation

Pinned to now stable v0.5 release with some fixes since previous review. This is a reopen of #18419 since that PR was orphaned (Sorry!).
The only SerenityOS-specific patch is support for `anon_create(2)` syscall and is handled upstream.

Hope one day we can run Serenity on it :D

Screenshot for the sake of it:

232634196-44009bd0-0b20-455f-8295-69f4cc7de31b

Any objections appreciated.

P.S. it seems the kernel scheduler is a bit unfair to IO tasks, some RVVM guests (Haiku bootloader) like to busy-wait meanwhile the IO threadpool (based on `pthread_cond`) is brutally slow... This doesn't occur on other platforms so someone might be interested in this.

### CODE REVIEW

File: Ports/rvvm/package.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='rvvm'
+version='0.5'
+archive_hash=3a1dbb91ad04f068078bc6c6c27cc5792eebc111907cb5a14bde158fe6e757c9
```
Comment (GMTA) [inline Ports/rvvm/package.sh:4]: Please put this string inside single quotes as well :^)

File: Ports/rvvm/package.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+files="https://github.com/LekKit/RVVM/archive/v$version.tar.gz rvvm.tar.gz $archive_hash"
```
Comment (GMTA) [inline Ports/rvvm/package.sh:5]:
```suggestion
files="https://github.com/LekKit/RVVM/archive/v${version}.tar.gz rvvm.tar.gz ${archive_hash}"
```

**Review by GMTA — CHANGES_REQUESTED** [review]

File: Ports/rvvm/package.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+auth_type='sha256'
+workdir="RVVM-${version}"
+depends=('sdl12-compat')
+
+build() {
+    run make OS=SerenityOS USE_SDL=1 USE_NET=1 GIT_COMMIT=76796ba all lib
+}
+
+install() {
+    run make OS=SerenityOS USE_SDL=1 USE_NET=1 GIT_COMMIT=76796ba DESTDIR="${DESTDIR}" install
```
Comment (GMTA) [inline Ports/rvvm/package.sh:15]: Something I missed in my last review: all these makeopts can go into `makeopts`, e.g.:
```bash
makeopts+=(
    'OS=SerenityOS',
    'USE_SDL=1',
    'USE_NET=1',
    'GIT_COMMIT=76796ba'
)
```

And then you can leave out your `install()` override, and your `build()` will become something like:
```bash
build() {
    run make "${makeopts[@]}" all lib
}
```

Comment (LekKit) [reply Ports/rvvm/package.sh:15]: Seems it doesn't. The Makefile is not prepared to understand such configuration where you magically swap target OSes and thinks you are crosscompiling for multiple targets.

image

Comment (GMTA) [reply Ports/rvvm/package.sh:15]: Ah you are right, it uses `installopts` which you can set to the same value as `makeopts`.
https://github.com/SerenityOS/serenity/blob/868a3912f470c57181ef19ce0f83351f1e2ae5b3/Ports/.port_include.sh#L478

Comment (GMTA) [reply Ports/rvvm/package.sh:15]: > Also, the makeopts apparently should not have comas in it

Those commas were a syntactical mistake by me, should indeed be left out :-)

Comment (LekKit) [reply Ports/rvvm/package.sh:15]: Does this look fine to you?

```
makeopts+=(OS=SerenityOS USE_SDL=1 USE_NET=1 GIT_COMMIT=76796ba all lib)
installopts+=(${makeopts[@]})
*no build/install stages at all*
```

Comment (GMTA) [reply Ports/rvvm/package.sh:15]: Yes, apart from the fact that I'd put those elements inside of quotes!

```
build_opts=(
    'GIT_COMMIT=76796ba'
    'OS=SerenityOS'
    'USE_NET=1'
    'USE_SDL=1'
)
makeopts+=("${build_opts[@]}" 'all' 'lib')
installopts+=("${build_opts[@]}")
```

Something like that maybe?

Comment (LekKit) [reply Ports/rvvm/package.sh]: CI is failing now :C

Comment (GMTA) [reply Ports/rvvm/package.sh]: > "the authorized way"

Feel free to document some of this! We are always in need of more concise contribution guidelines such as these :^)

With your latest changes, it seems that it builds and installs correctly!

**Comment by GMTA** [comment]
@LekKit FYI, the project fails to install on MacOS since `install` behaves differently:

```
[rvvm/install] [INFO] Installing to prefix /Users/jelle.raaijmakers/Projects/serenity/Build/x86_64/Root/usr/local
[rvvm/install] install: illegal option -- D
[rvvm/install] usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
[rvvm/install]                [-o owner] file1 file2
[rvvm/install]        install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
[rvvm/install]                [-o owner] file1 ... fileN directory
[rvvm/install]        install -d [-v] [-g group] [-m mode] [-o owner] directory ...
[rvvm/install] make: *** [install] Error 64
```

Since this looks like an upstream issue, I'm going to ignore it for this PR.

**Review by GMTA — APPROVED** [review]

**Comment by GMTA** [comment]
Whoops, I missed that there were 2 new commits. My bad.

https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md

> https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md

**Pull request merged**